tfexec/init.go

```diff
@@ -114,17 +122,27 @@ func (tf *Terraform) initCmd(ctx context.Context, opts ...InitOption) (*exec.Cmd
 	if c.fromModule != "" {
 		args = append(args, "-from-module="+c.fromModule)
 	}
-	if c.lockTimeout != "" {
-		args = append(args, "-lock-timeout="+c.lockTimeout)
+
+	// string opts removed in 0.15: pass if set and <0.15
+	err := tf.compatible(ctx, nil, tf0_15_0)
+	if err == nil {
+		if c.lockTimeout != "" {
+			args = append(args, "-lock-timeout="+c.lockTimeout)
+		}
 	}
 
 	// boolean opts: always pass
 	args = append(args, "-backend="+fmt.Sprint(c.backend))
 	args = append(args, "-get="+fmt.Sprint(c.get))
-	args = append(args, "-get-plugins="+fmt.Sprint(c.getPlugins))
-	args = append(args, "-lock="+fmt.Sprint(c.lock))
 	args = append(args, "-upgrade="+fmt.Sprint(c.upgrade))
-	args = append(args, "-verify-plugins="+fmt.Sprint(c.verifyPlugins))
+
+	// boolean opts removed in 0.15: pass if <0.15
+	err = tf.compatible(ctx, nil, tf0_15_0)
```

Comment:
Nitpick: Should we also gate these flags on the other end (minimum version), similar to how we gate `fmt`?

Reply:
Yes, added https://github.com/hashicorp/terraform-exec/issues/121 as it's a somewhat bigger job to go through and find the minimum versions for all flags, which might as well be done all at once. Merging this PR now so we can see if the next nightly passes.